core/src/test/java/io/github/tcdl/configuration/ConfigurationUtilTest.java

```diff
@@ -0,0 +1,158 @@
+package io.github.tcdl.configuration;
+
+import com.typesafe.config.Config;
+import io.github.tcdl.config.ConfigurationUtil;
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+import java.util.Optional;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.assertFalse;
+import static org.junit.Assert.assertTrue;
+import static org.mockito.Mockito.when;
+
+/**
+ * Created by rdrozdov-tc on 6/11/15.
+ */
+@RunWith(MockitoJUnitRunner.class)
+public class ConfigurationUtilTest {
+
+    @Mock
+    private Config config;
+
+    @Before
+    public void setUp() {
+    }
+
+    @Test
+    public void testGetBooleanExists() {
+        String param = "parameter.boolean";
+        Boolean paramValue = true;
+        Boolean fallback = true;
+
+        when(config.hasPath(param)).thenReturn(true);
+        when(config.getBoolean(param)).thenReturn(paramValue);
+
+        Boolean value = ConfigurationUtil.getBoolean(config, param, fallback);
+
+        assertEquals(paramValue, value);
```

Comment:
I would use assertTrue here. but actually i would like to see 3 cases here:

1.
when(config.hasPath(param)).thenReturn(true);
Boolean value = ConfigurationUtil.getBoolean(config, param, fallback);
verify(config).getBoolean(eq(param))

2. when(config.hasPath(param)).thenReturn(false);
Boolean value = ConfigurationUtil.getBoolean(config, param, fallback);
verify(config, never()).getBoolean(eq(param))

3.  when(config.hasPath(param)).thenReturn(true);
 Boolean value = ConfigurationUtil.getBoolean(config, param, fallback);    
verify(config, never()).getBoolean(eq(param))
 assertEquals(fallback, value);

Or even combine 2 and 3 into one case